pyanaconda/modules/common/task/__init__.py

```diff
@@ -67,3 +69,27 @@ def _callback():
 
     task_proxy.Stopped.connect(_callback)
     task_proxy.Start()
+
+
+def wait_for_task(task_proxy, timeout=math.inf):
+    """Wait for an existing and running task with optional timeout.
+
+    If this returns True and the task is finished, you should call the Finish method of the remote
+    task and handle the remote errors. If this returns False, either
+
+    :param task_proxy: a proxy of the remote task
+    :param float timeout: stop waiting after this time in seconds
+    :return bool: did the task finish or not
+    """
+    start = perf_counter()
+    end = start + timeout
+
+    while task_proxy.IsRunning and perf_counter() <= end:
+        sleep(0.1)
+
+    if not task_proxy.IsRunning:
+        task_proxy.Finish()
+        return True
+
+    else:
+        return False
```

Comment:
I prefer to raise and handle the `TimeoutError` exception in these cases. It somehow feels more standard, but this solution is also fine.

Reply:
Let's do that, I have no preference.

I also wondered if there should be a handler to attach in case the Task is not finished. But at the same time, if the finished/unfinished state is communicated in other ways, I'm not sure what value that could bring.